### PR TITLE
feature: login functionality

### PR DESCRIPTION
**Pull request by MohitSJadhav**

### CODE REVIEW

File: src/pages/SignIn.js [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+                        <TextField
+                            required
+                            fullWidth
+                            id="email"
+                            label="Email Address"
+                            name="email"
+                            autoComplete="email"
+                            value={email}
+                            helperText="Valid hawk or gmail id"
+                            inputProps={{ pattern: "^[a-zA-Z0-9]+@(hawk.iit.edu|gmail.com)$" }}
```
Comment (anushasonte) [inline src/pages/SignIn.js]: Allow "." in the pattern match. [I got error on my email: anusha.sonte@gmail.com]

File: src/pages/SignUp.js [hunk 2]
```diff
@@ -75,77 +104,185 @@ export default function SignUp() {
...
+                  <TextField
+                    required
+                    fullWidth
+                    id="email"
+                    label="Email Address"
+                    name="email"
+                    autoComplete="email"
+                    value={email}
+                    helperText="Valid hawk or gmail id"
+                    inputProps={{ pattern: "^[a-zA-Z0-9]+@(hawk.iit.edu|gmail.com)$" }}
```
Comment (anushasonte) [inline src/pages/SignUp.js]: Allow "." in the pattern match.

File: src/pages/SignUp.js [hunk 3]
```diff
@@ -75,77 +104,185 @@ export default function SignUp() {
...
+                      required
+                    >
+                      {[...Array(41).keys()].map((value) => {
+                        value += 10;
+                        return <MenuItem key={value} value={value}>{value}</MenuItem>
+                      })}
+                    </Select>
+                  </FormControl>
+                </Grid>
+
```
Comment (anushasonte) [inline src/pages/SignUp.js:225]: Can we have gender, userType and age displayed in the same line?

File: src/pages/SignUp.js [hunk 4]
```diff
@@ -75,77 +104,185 @@ export default function SignUp() {
...
+                      <MenuItem value=""><em>None</em></MenuItem>
+                      <MenuItem value={"User"}>User</MenuItem>
+                      <MenuItem value={"Event Provider"}>Event Provider</MenuItem>
+                      <MenuItem value={"System Admin"}>System Admin</MenuItem>
+                      <MenuItem value={"Content Moderator"}>Content Moderator</MenuItem>
+                    </Select>
+                  </FormControl>
+                </Grid>
+
+                {/* Age */}
```
Comment (anushasonte) [inline src/pages/SignUp.js:203]: We need to allow the value of Age to be 50+ as well. Probably prof is 50+. (lets keep it as text with input validation like : Age cant be greater than 110 or less than 5)

File: src/pages/SignUp.js [hunk 5]
```diff
@@ -75,77 +104,185 @@ export default function SignUp() {
...
+                      labelId="gender-label"
+                      id="gender"
+                      name='gender'
+                      onChange={(event, data) => { setGender(event.target.value) }}
+                      label="Gender"
+                      fullWidth
+                      value={gender}
+                      required
+                    >
+                      <MenuItem value=""><em>None</em></MenuItem>
```
Comment (anushasonte) [inline src/pages/SignUp.js]: Remove 'None' from gender values and and 'Other.'

File: src/pages/SignUp.js [hunk 6]
```diff
@@ -75,77 +104,185 @@ export default function SignUp() {
...
+                      labelId="user-label"
+                      id="user"
+                      name='user'
+                      value={user}
+                      onChange={(event, data) => { setUserType(event.target.value) }}
+                      label="User Type"
+                      fullWidth
+                      required
+                    >
+                      <MenuItem value=""><em>None</em></MenuItem>
```
Comment (anushasonte) [inline src/pages/SignUp.js]: Remove 'None' from user type and let 'User' be the default type.

File: src/pages/SignUp.js [hunk 7]
```diff
@@ -75,77 +104,185 @@ export default function SignUp() {
...
+                      <MenuItem value={"Male"}>Male</MenuItem>
+                      <MenuItem value={"Female"}>Female</MenuItem>
+                    </Select>
+                  </FormControl>
+                </Grid>
+
+                {/* User Type */}
+                <Grid item xs={12} sm={6}>
+                  <FormControl sx={{ m: 1, minWidth: 100 }}>
+                    <InputLabel id="user-select-autowidth-label">User Type</InputLabel>
```
Comment (anushasonte) [inline src/pages/SignUp.js]: Lets call this 'Role' as we already have a 'user' type.

File: src/pages/SignIn.js (hunk 1 above)
Comment (MohitSJadhav) [reply src/pages/SignIn.js]: Done

**Pull request merged**